### DISCUSSION TITLE
ESP32 BLE/WiFi coexistence

### DISCUSSION THREAD

**Discussion by real-tintin** [zephyrproject-rtos/zephyr#80899] [Q&A]
Hi,

I'm wondering if there are any plans for implementing BLE/WiFi coexistence for ESP32? The ESP32_WIFI_SW_COEXIST_ENABLE is config is already declared but not used, clearly states: "Software controls WiFi/Bluetooth coexistence. Not supported yet."

I don't how much effort it would be to implement. In terms of average power usage our findings are:

- Only BLE: 30 mA
- Only WiFi (modem sleep mode): 60 mA
- BLE & WiFi: 100 mA

all at 3V.

I guess that when both BLE and WiFi is used, the radio is at max and that there is no overlap i.e., the BLE and WiFi drivers operate independently?

**Comment by rftafas** [comment]
Hi,

coexistence is already working, but it is the hardware based one, which cannot handle high traffic scenarios. Yes, there are plans for enabling the software based, but since the hardware one covers the majority of applications, it keeps getting delayed.

Not sure if I got the last question right. They are independent but do interact. Also, sub-bands don't overlap on Wi-Fi and BT. Same cannot be said between Wi-Fi and 15.4

**Reply by real-tintin** [reply]
Ok! Thanks for the clarification.

On a side note, we managed to reduce consumption by enabling BT modem sleep. I have created a PR for this. https://github.com/zephyrproject-rtos/hal_espressif/pull/361
